libs/application/ui-shell/src/reducer/reducerUtils.ts

```diff
@@ -29,53 +30,141 @@ import {
+  screen: FormScreen,
+  answers: FormValue,
+  checkIfPartlyAnswered = false,
+) => {
+  if (!screenRequiresAnswer(screen)) {
+    return true
+  }
+
+  if (!screen.id) {
+    return false
+  }
+
+  const answer = getValueViaPath(answers, screen.id)
+  const answerHasValue = answer !== undefined && answer !== null
+
+  if (screen.type === FormItemTypes.REPEATER) {
+    // We do not need to check all individual screens here like we do
+    // with multi field since the screens are rendered individually
+    // and will be checked individually
+    return isArray(answer) && answer.length > 0
+  } else if (screen.type === FormItemTypes.MULTI_FIELD) {
+    let numberOfAnswers = 0
+    let numberOfNotNavigable = 0
+
+    for (const subScreen of screen.children) {
+      if (!subScreen.isNavigable) {
+        numberOfNotNavigable += 1
+      } else if (screenHasBeenAnswered(subScreen, answers)) {
+        numberOfAnswers += 1
+      }
+    }
+
+    if (checkIfPartlyAnswered) {
+      return numberOfAnswers > 0
+    }
+
+    return numberOfAnswers + numberOfNotNavigable === screen.children.length
+  }
+
+  return answerHasValue
+}
+
 export const findCurrentScreen = (
   screens: FormScreen[],
   answers: FormValue,
 ): number => {
   let currentScreen = 0
-  let missingAnswerBeforeCurrentIndex = false
-  let currentAnswerIndex = 0
+  let missingAnswer = false
+  let lastAnswerIndex = -1
 
-  screens.forEach((screen, index) => {
-    if (screen.type === FormItemTypes.MULTI_FIELD) {
-      let numberOfAnsweredQuestionsInScreen = 0
+  for (let index = 0; index < screens.length; ) {
+    const screen = screens[index]
 
-      screen.children.forEach((field) => {
-        if (getValueViaPath(answers, field.id) !== undefined) {
-          numberOfAnsweredQuestionsInScreen++
-        }
-      })
+    const requiresAnswer = screenRequiresAnswer(screen)
+
+    if (!requiresAnswer) {
+      let skipLength = 1
+
+      if (screen.type === FormItemTypes.REPEATER) {
+        skipLength = screen.children.length + 1
+      }
+
+      currentScreen = index + skipLength
+      index = index + skipLength
 
-      if (numberOfAnsweredQuestionsInScreen === screen.children.length) {
+      continue
+    }
+
+    const hasBeenAnswered = screenHasBeenAnswered(screen, answers)
+
+    if (screen.type === FormItemTypes.MULTI_FIELD) {
+      if (hasBeenAnswered) {
+        lastAnswerIndex = index
         currentScreen = index + 1
-      } else if (numberOfAnsweredQuestionsInScreen > 0) {
+      } else if (screenHasBeenAnswered(screen, answers, true)) {
+        // Checks if screen has been partly answered (at least one answer found)
+        lastAnswerIndex = index
         currentScreen = index
+      } else {
+        // Did not answer any question
+        missingAnswer = true
       }
     } else if (screen.type === FormItemTypes.REPEATER) {
-      if (getValueViaPath(answers, screen.id) !== undefined) {
+      if (hasBeenAnswered) {
+        lastAnswerIndex = index
```

Comment:
How does the screens array look here with repeaters? Are the sub-screen flattened so it can "find" and continue editing the last entry in the repeater? Or will it evaluate the whole repeater as one screen, just checking that it contains a non-empty array?